src/main/java/org/symphonyoss/symphony/messageml/elements/UIAction.java

```diff
@@ -0,0 +1,167 @@
+  }
+
+  @Override
+  protected void buildAttribute(MessageMLParser parser, org.w3c.dom.Node item) throws InvalidInputException {
+    switch (item.getNodeName()) {
+      case ACTION_ATTR:
+      case TRIGGER_ATTR:
+      case USER_IDS_ATTR:
+      case STREAM_ID_ATTR:
+      case SIDE_BY_SIDE_ATTR:
+        if (this.format != FormatEnum.MESSAGEML) {
+          throwInvalidInputException(item);
+        }
+        setAttribute(item.getNodeName(), getStringAttribute(item));
+        break;
+      default:
+        throwInvalidInputException(item);
+    }
+  }
+
+  @Override
+  public void validate() throws InvalidInputException {
+    assertUIActionAllowedChildren();
+
+    assertAttributeNotBlank(ACTION_ATTR);
+    assertAttributeValue(ACTION_ATTR, Collections.singleton(ALLOWED_ACTION));
+    // validate attributes specific to the 'open-im' action
+    validateOpenChatActionAttributes();
+
+    if(getAttribute(TRIGGER_ATTR) != null) {
+      assertAttributeValue(TRIGGER_ATTR, Collections.singleton(DEFAULT_TRIGGER));
+    }
+  }
+
+  private void validateOpenChatActionAttributes() throws InvalidInputException {
+    if(getAttribute(SIDE_BY_SIDE_ATTR) != null) {
+      assertAttributeIsBoolean(SIDE_BY_SIDE_ATTR);
+    }
+    if (getAttribute(USER_IDS_ATTR) != null && getAttribute(STREAM_ID_ATTR) != null) {
+      throw new InvalidInputException("Only one between \"stream-id\" and \"user-ids\" is allowed");
+    }
+    if (getAttribute(USER_IDS_ATTR) == null && getAttribute(STREAM_ID_ATTR) == null) {
+      throw new InvalidInputException("At least one between \"stream-id\" and \"user-ids\" should be present");
+    }
+    validateUserIdsAttribute();
+  }
+
+  private void assertUIActionAllowedChildren() throws InvalidInputException {
+    List<Class<? extends Element>> allowedChildren = Arrays.asList(Button.class, UIAction.class);
+    assertContentModel(allowedChildren);
+    assertContainsAlwaysChildOfType(allowedChildren);
+  }
+
+  @Override
+  void asPresentationML(XmlPrintStream out, MessageMLContext context) {
+    Map<String, String> presentationAttrs = buildAUIActionAttributes();
+    out.openElement(getPresentationMLTag(), presentationAttrs);
+    for (Element child : getChildren()) {
+      child.asPresentationML(out, context);
+    }
+    out.closeElement();
+  }
+
+  @Override
+  public String getPresentationMLTag() {
+    return PRESENTATIONML_TAG;
+  }
+
+  private void validateUserIdsAttribute() throws InvalidInputException {
+    String userIds = getAttribute(USER_IDS_ATTR);
+    List<Long> userIdsList = new ArrayList<>();
+    if (userIds != null) {
+      try {
+        userIdsList = MAPPER.readValue(userIds, MAPPER.getTypeFactory().constructCollectionType(List.class, Long.class));
+      } catch (JsonProcessingException e) {
+        throw new InvalidInputException("Attribute \"user-ids\" contains an unsupported format, should be an array of user ids");
+      }
+    }
+    if(userIdsList.size() > MAX_USER_IDS){
```

Comment:
we could move the if after the read value, this way we can avoid creating an empty array list (not very important but  it will be simpler to read IMHO)

Reply:
Done in https://github.com/symphonyoss/messageml-utils/pull/262/commits/8fed1fb980d6bd286508fce633da38572d7a1595